### PR TITLE
Add vic-machine configure ops and tprint support

### LINKED ISSUE #4495: vic-machine reconfigure: Update vCenter server credentials
**User Statement:**

As a VIC operation user, I need the ability to update the server credentials (username, password/SSO token and server thumbprint) for the appliance, in case the vCenter server credentials have been changed. See one example here #1879.

**Details:**
- supported operations:
  - update username
  - update password
  - update thumbprint
- hot update:
  - no; need to restart the VCH to make the changes take effect.
- input/output data format:
  - input: plain text, e.g., `--username=new-username -- password=new-pwd -- thumbprint=new-thumbprint`
  - output: N/A
- impact on VCH/containerVMs:
  - VCH: The changes will be persisted to the vmx file of the VCH. Then the portlayer will need to reload the VCH configure from the vmx file during VCH restart. 
  - containerVMs: no
- is upgrade needed: 
   -  no, since no VCH configuration changes. 

**Acceptance Criteria:**
-  add the ability to update the server credentials such as username, password/SSO token and server thumbprint though `vic-machine reconfigure`; and the changes will be applied after VCH restart
- add relevant test cases

### PR DESCRIPTION
**Pull request by anchal-agrawal**
Fixes #4495

### CODE REVIEW

File: lib/install/data/data.go [hunk 1]
```diff
@@ -253,6 +252,14 @@ func (d *Data) CopyNonEmpty(src *Data) error {
 		}
 	}
 
+	if src.OpsCredentials.IsSet {
+		d.OpsCredentials = src.OpsCredentials
+	}
+
+	if src.Target.Thumbprint != "" {
```
Comment (emlin) [inline lib/install/data/data.go]: feels like you can compare if they are same here

Comment (emlin) [inline lib/install/data/data.go]: or just copy

**Comment by emlin** [comment]
@anchal-agrawal did you try to test using --force? suppose user does not have to specify thumbprint with --force specified.

**Comment by anchal-agrawal** [comment]
@emlin Good catch. If `--force` is specified and the input thumbprint is empty, it gets overwritten in the validator code. This change remains unaffected, so the following workflow works well:

* Create VCH with thumbprint `foo`
* Update thumbprint to `bar`
* `inspect --conf` - shows old thumbprint `foo`
* `configure --force ...` (no thumbprint specified)
* `inspect --conf` - shows new thumbprint `bar`

File: lib/install/validate/validator.go [hunk 2]
```diff
@@ -417,16 +417,18 @@ func (v *Validator) managedbyVC(ctx context.Context) {
...
 
 	// check target with ops credentials
 	u := input.Target.URL
 
-	conf.Username = input.OpsUser
-	conf.Token = *input.OpsPassword
+	if input.OpsCredentials.OpsUser != nil {
+		conf.Username = *input.OpsCredentials.OpsUser
+	}
+	conf.Token = *input.OpsCredentials.OpsPassword
```
Comment (emlin) [inline lib/install/validate/validator.go]: opspassword could be nil as well

Comment (anchal-agrawal) [inline lib/install/validate/validator.go]: That's taken care of at the top of the function 😃

Comment (emlin) [inline lib/install/validate/validator.go]: hmm.. no, actually you tested the password, but do not for the user, which is slightly confusing.
I think you might just do the same thing for the opsuser name, how do you think?

Comment (anchal-agrawal) [inline lib/install/validate/validator.go]: Sounds good, fixed.

**Pull request merged**